warehouse/age-off-utils/src/main/java/datawave/age/off/util/AgeOffRuleFormatter.java

```diff
@@ -51,8 +51,13 @@ void format(Writer writer) throws IOException {
         writer.close();
     }
 
+    private static void increment() {
+        index++;
+    }
+
     private AgeOffRuleLoader.RuleConfig createRuleConfig(AgeOffRuleConfiguration configuration) throws IOException {
-        AgeOffRuleLoader.RuleConfig ruleConfig = new AgeOffRuleLoader.RuleConfig(this.configuration.getFilterClass().getName(), index++);
+        AgeOffRuleLoader.RuleConfig ruleConfig = new AgeOffRuleLoader.RuleConfig(this.configuration.getFilterClass().getName(), index);
```

Comment:
I know this was sorta the quick fix to get thru the sonarqube complaint but I almost feel like we're hiding the problem.

Can we make `createRuleConfig()` static instead? It has some instance references to `configuration` but we actually pass `configuration` in already - so I think we can make it just use the passed in one.

@matthpeterson Does that sound reasonable?

Reply:
Looks like we'd need to pass indent in to `createRuleConfig()` and then to `buildMatchPattern()` as well.

Reply:
reworked it as you described.